screwdriver/api/api.go

```diff
@@ -33,102 +38,212 @@ type ResponseError struct {
+	c := newClient(sdAPI, sdToken)
+	return API(c)
 }
 
-func newClient(sdAPI, sdToken string) (*client, error) {
-	c := &client{
+func newClient(sdAPI, sdToken string) *client {
+	return &client{
 		baseURL: sdAPI,
 		jwt:     sdToken,
 		client:  &http.Client{Timeout: timeoutSec * time.Second},
 	}
-	return c, nil
 }
 
-func handleResponse(res *http.Response) ([]byte, error) {
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Reading response Body from Screwdriver API: %v", err)
-	}
-
-	switch res.StatusCode / 100 {
+func handleResponse(bodyBytes []byte, statusCode int) ([]byte, error) {
+	switch statusCode / 100 {
 	case 2:
-		return body, nil
+		res := new(util.CommandSpec)
+		err := json.Unmarshal(bodyBytes, res)
+		if err != nil {
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
+		}
+		return bodyBytes, nil
 	case 4:
 		res := new(ResponseError)
-		err = json.Unmarshal(body, res)
+		err := json.Unmarshal(bodyBytes, res)
 		if err != nil {
-			return nil, fmt.Errorf("Unparseable error response from Screwdriver API: %v", err)
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
 		}
 		return nil, res
 	case 5:
-		return nil, fmt.Errorf("%v: Screwdriver API has internal server error", res.StatusCode)
+		return nil, fmt.Errorf("Screwdriver API has internal server error: statusCode=%d", statusCode)
 	default:
-		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API")
+		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API: Statuscode=%d", statusCode)
 	}
 }
 
 // GetCommand returns Command from Screwdriver API
-func (c client) GetCommand(smallSpec *Command) (*Command, error) {
-	cmd := new(Command)
+func (c client) GetCommand(smallSpec *util.CommandSpec) (*util.CommandSpec, error) {
 	uri, err := url.Parse(c.baseURL)
 	if err != nil {
-		return nil, fmt.Errorf("The base Screwdriver API is invalid %q", c.baseURL)
+		return nil, fmt.Errorf("Failed to parse URL on GET: %v", err)
 	}
 	uri.Path = path.Join(uri.Path, "commands", smallSpec.Namespace, smallSpec.Name, smallSpec.Version)
-	req, err := http.NewRequest("GET", uri.String(), strings.NewReader(""))
+
+	// Request to api
+	const contentType = "application/json"
+	// No payload
+	payload := bytes.NewBuffer([]byte(""))
+
+	responseSpec, err := c.httpRequest("GET", uri.String(), contentType, payload)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create request about command to Screwdriver API: %v", err)
+		return nil, err
 	}
 
-	req.Header.Set("Content-type", "application/json")
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.jwt))
-	res, err := c.client.Do(req)
+	return responseSpec, nil
+}
+
+func specToPayloadBytes(commandSpec *util.CommandSpec) (specBytes []byte, err error) {
+	// Generate yaml payload bytes
+	commandSpecBytes, err := json.Marshal(&commandSpec)
```

Comment:
Maybe I think you do not need to pass `&commandSpe`c to` json.Marshal` method. 
Maybe you just need to send `commandSpec.
https://blog.golang.org/json-and-go

Reply:
Thank you! I've fixed. I also fixed other similar place.